nodestream/pipeline/writers.py

```diff
@@ -17,13 +18,22 @@ async def handle_async_record_stream(
         self, record_stream: AsyncGenerator[Any, Any]
     ) -> AsyncGenerator[Any, Any]:
         async for record in record_stream:
-            await self.write_record(record)
+            if record is Flush:
```

Comment:
Just to understand. The extractor generates Flush as an EOF marker of sorts. Then the writer recognizes it and tells  the ingestion strategy to flush. Initial thought is why not put this conditional on the ingestion strategy and make Flush an ingestible?

If my understanding is correct, flushing is a side effect of an ingestion strategy that is enabled by the injection of the EOF marker by the extractor. In this case writer is an auxiliary when it doesn't need to be.

Reply:
My interpretation of this change was that the writer would be able to flush to a downstream step like another writer or another ingestion component. Like suppose there is something complicated like:

Extractor:
Transformer 1:
Writer 1:
ConcurrentTransformer 2: 
Ingestion:
Writer 2:

Something like the flush would be necessary for ConcurrentTransformer 2 to know that it shouldn't expect any new data from the upstream service and shouldn't spawn new processes.
Or if we were just writing to writer 1 and writer 2, it lets it know that it shouldn't wait for the pending_records to reach the batch size limit. Like pending_records = 500 < 1000, the writer will just write what we have.

Reply:
@angelosantos4 has essentially the right idea. Does that explanation make sense @yasonk ?

Reply:
@angelosantos4 @zprobst I think I understand, but my comment stems from the lower level code. Where the writer simply calls `ingestion_strategy.flush()`.
I guess my understanding wrong that flushing is a feature of the ingestion_strategy.

Reply:
I guess the code I was looking at already sort of exists, but not sure how it is hooked up:
https://github.com/nodestream-proj/nodestream/blob/a2ece8a6741cf4e7a488ef3d21149d8173e6ba31/nodestream/databases/writer.py#L41

Reply:
Capturing an offline conversation for the record:

Essentially this conversation boiled down to:

- Handling of `Flush` did not belong in `GraphDatabaseWriter`
- Q: Does it belong in `Writer` or `IngestStrategy`?
- A: It has to belong in `Writer` in the current state because `Writer` subclasses don't strictly operate on `ingestible` things.